src/Microsoft.OpenApi.Readers/YamlConverter.cs

```diff
@@ -0,0 +1,133 @@
+        public static IEnumerable<JsonNode> ToJsonNode(this YamlStream yaml)
+        {
+            return yaml.Documents.Select(x => x.ToJsonNode());
+        }
+
+        /// <summary>
+        /// Converts a single YAML document to a <see cref="JsonNode"/>.
+        /// </summary>
+        /// <param name="yaml">The YAML document.</param>
+        /// <returns>A `JsonNode` representative of the YAML document.</returns>
+        public static JsonNode ToJsonNode(this YamlDocument yaml)
+        {
+            return yaml.RootNode.ToJsonNode();
+        }
+
+        /// <summary>
+        /// Converts a single YAML node to a <see cref="JsonNode"/>.
+        /// </summary>
+        /// <param name="yaml">The YAML node.</param>
+        /// <returns>A `JsonNode` representative of the YAML node.</returns>
+        /// <exception cref="NotSupportedException">Thrown for YAML that is not compatible with JSON.</exception>
+        public static JsonNode ToJsonNode(this YamlNode yaml)
+        {
+            return yaml switch
+            {
+                YamlMappingNode map => map.ToJsonObject(),
+                YamlSequenceNode seq => seq.ToJsonArray(),
+                YamlScalarNode scalar => scalar.ToJsonValue(),
+                _ => throw new NotSupportedException("This yaml isn't convertible to JSON")
+            };
+        }
+
+        /// <summary>
+        /// Converts a single JSON node to a <see cref="YamlNode"/>.
+        /// </summary>
+        /// <param name="json"></param>
+        /// <returns></returns>
+        /// <exception cref="NotSupportedException"></exception>
+        public static YamlNode ToYamlNode(this JsonNode json)
+        {
+            return json switch
+            {
+                null => null,
+                JsonObject obj => obj.ToYamlMapping(),
+                JsonArray arr => arr.ToYamlSequence(),
+                JsonValue val => val.ToYamlScalar(),
+                _ => throw new NotSupportedException("This isn't a supported JsonNode")
+            };
+        }
+
+        private static JsonObject ToJsonObject(this YamlMappingNode yaml)
+        {
+            var node = new JsonObject();
+            foreach (var keyValuePair in yaml)
+            {
+                var key = ((YamlScalarNode)keyValuePair.Key).Value!;
+                node[key] = keyValuePair.Value.ToJsonNode();
+            }
+
+            return node;
+        }
+
+        private static YamlMappingNode ToYamlMapping(this JsonObject obj)
+        {
+            return new YamlMappingNode(obj.ToDictionary(x => (YamlNode)new YamlScalarNode(x.Key),
+                x => x.Value!.ToYamlNode()));
+        }
+
+        private static JsonArray ToJsonArray(this YamlSequenceNode yaml)
+        {
+            var node = new JsonArray();
+            foreach (var value in yaml)
+            {
+                node.Add(value.ToJsonNode());
+            }
+
+            return node;
```

Comment:
```suggestion
            return new JsonArray(yaml.Select(static x => x.ToJsonNode()).ToArray());
```

Reply:
Your suggestion allocates an extra array.

Reply:
right, sorry I meant this

```suggestion
            return new JsonArray(yaml.Select(static x => x.ToJsonNode()));
```